addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,356 @@
+        params[questionId] = formattedDate;
+        return params;
+    },
+
+    /**
+    *   Prepare choice answer before submitting form.
+    *   If the answer is not the 'comment selection' (=Other), calls the _prepareSubmitAnswer method to add the answer to the params
+    *   If there is a comment linked to that question, calls the _prepareSubmitComment method to add the comment to the params
+    */
+    _prepareSubmitChoices: function(self, params, $parent, questionId, selector) {
+        $parent.find(selector).each(function () {
+            if (this.value != '-1') {
+                params = self._prepareSubmitAnswer(params, questionId, this.value);
+            }
+        });
+        params = self._prepareSubmitComment(self, params, $parent, questionId, false);
+        return params;
+    },
+
+    /**
+    *   Prepare answer before submitting form if question type is matrix.
+    *   This method regroups answers by question and by row to make a object like :
+    *   params = { 'matrixQuestionId' : { 'rowId1' : [colId1, colId2,...], 'rowId2' : [colId1, colId3, ...] } }
+    */
+    _prepareSubmitAnswerMatrix: function(params, questionId, rowId, colId) {
+        var value = questionId in params ? params[questionId] : {};
+        if (rowId in value) {
+            value[rowId].push(colId);
+        } else {
+            value[rowId] = [colId];
+        }
+        params[questionId] = value;
+        return params;
+    },
+
+    /**
+    *   Prepare answer before submitting form (any kind of answer - except Matrix -).
+    *   This method regroups answers by question.
+    *   Lonely answer are directly assigned to questionId. Multiple answers are regrouped in an array:
+    *   params = { 'questionId1' : lonelyAnswer, 'questionId2' : [multipleAnswer1, multipleAnswer2, ...] }
+    */
+    _prepareSubmitAnswer: function(params, questionId, value) {
+        if (questionId in params) {
+            if (params[questionId].constructor === Array) {
+                params[questionId].push(value);
+            } else {
+                params[questionId] = [params[questionId], value];
+            }
+        } else {
+            params[questionId] = value;
+        }
+        return params;
+    },
+
+    /**
+    *   Prepare comment before submitting form.
+    *   This method extract the comment, encapsulate it in a dict and calls the _prepareSubmitAnswer methods
+    *   with the new value. At the end, the result looks like :
+    *   params = { 'questionId1' : {'comment': commentValue}, 'questionId2' : [multipleAnswer1, {'comment': commentValue}, ...] }
+    */
+    _prepareSubmitComment: function(self, params, $parent, questionId, isMatrix) {
+        $parent.find('textarea').each(function () {
+            if (this.value) {
+                var value = {'comment': this.value};
+                if (isMatrix) {
+                    params = self._prepareSubmitAnswerMatrix(params, questionId, this.name, value);
```

Comment:
So, according the `_prepareSubmitAnswerMatrix` docstring, `this.name` is 'rowId' and  `value` is 'colId' ? :scream: